packages/neo4j/src/utils.ts

```diff
@@ -1,93 +1,65 @@
-import neo4j from "neo4j-driver"
+import type { Session } from "neo4j-driver"
+import { isInt, integer } from "neo4j-driver"
 
-const neo4jToSafeNumber = (x: typeof neo4j.Integer) => {
-  if (!neo4j.isInt(x)) {
-    return x
-  }
-  if (neo4j.integer.inSafeRange(x)) {
-    return x.toNumber()
-  } else {
-    return x.toString()
-  }
-}
+// https://github.com/honeinc/is-iso-date/blob/master/index.js
+const isoDateRE =
+  /(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z))|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d([+-][0-2]\d:[0-5]\d|Z))|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d([+-][0-2]\d:[0-5]\d|Z))/
 
-// Function to transform date values from neo4j DateTime to JS Date object.
-const neo4jDateToJs = (value: typeof neo4j.DateTime | null) => {
-  if (!value || !neo4j.temporal.isDateTime(value)) {
-    return value
-  }
-  return new Date(value.toString())
+function isDate(value: any) {
+  return value && isoDateRE.test(value) && !isNaN(Date.parse(value))
```

Comment:
This is the only part I did not really like, but I cannot see another way of solving this.

Neo4j does not accept `Date`, so they are saved as ISO strings.

When reading the values ideally these would be converted back to `Date`, but Neo4j cannot do that. We have to either detect ISO Date strings somehow. This is the safest way I could do that.